### PR TITLE
SQL Task 01

### PR DESCRIPTION
**Pull request by DimitarL**

### CODE REVIEW

File: SQL_testing.php [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        {
+            $errors['$text']="$text полето трябва да бъде по-малко от $size символа";
+        }
+    }
+	
+	if($_POST)
+	{		
+		validateField('title', $title_Max_Length, $errors);
+        validateField('description', $lecturer_Max_Length, $errors);
+        validateField('lecturer', $description_Max_Length, $errors);
```
Comment (tantanitaa) [inline SQL_testing.php:27]: Please fix identation.

File: SQL_testing.php [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+			$lecturer = $_POST['lecturer'];
+			
+			$conn = new PDO('mysql:host=localhost;dbname=www;charset=utf8', 'root', '');
+			if($conn === false)
+			{
+				echo"ERROR: Could not connect.";
+			}
+			else
+			{
+				$insertion = $conn->prepare("INSERT INTO electives(title,description,lecturer) VALUES ('$title', '$description', '$lecturer')");
```
Comment (tantanitaa) [inline SQL_testing.php]: Maybe you can extract the sql query in a separate string variable for better readability.

File: SQL_testing.php [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        validateField('description', $lecturer_Max_Length, $errors);
+        validateField('lecturer', $description_Max_Length, $errors);
+		
+		if(count($errors) == 0)
+		{
+			$title = $_POST['title'];
+			$description = $_POST['description'];
+			$lecturer = $_POST['lecturer'];
+			
+			$conn = new PDO('mysql:host=localhost;dbname=www;charset=utf8', 'root', '');
```
Comment (tantanitaa) [inline SQL_testing.php]: You can create a separate file dedicated to database connection like here: https://github.com/aprazova/WEB2020-php/tree/sql_module/SQL_module/SubjectForm

File: SQL_testing.html [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+			<input type="text" id="title" name="title"><br>
+			<label id="invalidTitle"></label><br>
+			
+			<label for="description">Описание: </label>
+			<input type="text" id="description" name="description"><br>
+			<label id="invalidDescription"></label><br>
+			
+			<label for="lecturer">Преподавател: </label>
+			<input type="text" id="lecturer" name="lecturer"><br>
+			<label id="invalidLecturer"></label><br>
```
Comment (tantanitaa) [inline SQL_testing.html]: You can put `required` attribute in input tags just to be sure they will be filled by the user and not left empty.

File: SQL_testing.html [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+		<meta charset="UTF-8"/>
+		<title>JAVA_01</title>
+		
+		<style>
+			body
+			{
+				border: 2px solid black;
+				text-align: center;
+			}
+		</style>
```
Comment (tantanitaa) [inline SQL_testing.html]: Internal styles are not a good practice. You can create a separate file for css only.

File: SQL_testing.html [hunk 6]
```diff
@@ -0,0 +1,32 @@
+<!DOCTYPE html>
+<html lang="bg">
+	<head>
+		<meta charset="UTF-8"/>
+		<title>JAVA_01</title>
```
Comment (tantanitaa) [inline SQL_testing.html]: Why Java? :D

File: SQL_testing.php [hunk 7]
```diff
@@ -0,0 +1,50 @@
+<?php
+	$title_Max_Length = 128;
+	$lecturer_Max_Length = 128;
+	$description_Max_Length = 1024;
```
Comment (tantanitaa) [inline SQL_testing.php:6]: I like that you extract these as variables - it's easier to change and maintain.

**Review by tantanitaa — COMMENTED** [review]
Good job overall 👍

File: SQL_testing.php (hunk 2 above)
Comment (martinely95) [reply SQL_testing.php]: Currently the code is vulnerable to sql injection as you are preparing the whole expression. You may want to look here for reference: https://github.com/tancheto/WEB2020-php/pull/8/files The values are added after the sql is prepared.

File: SQL_testing.php (hunk 3 above)
Comment (DimitarL) [reply SQL_testing.php]: Fixed

File: SQL_testing.php (hunk 2 above)
Comment (DimitarL) [reply SQL_testing.php]: Fixed

**Comment by martinely95** [comment]
Better!

**Pull request merged**